assets/js/components/product-attribute-control/index.js

```diff
@@ -15,24 +15,16 @@ import { withAttributes } from '../../hocs';
 import ErrorMessage from '../api-error-placeholder/error-message.js';
 import './style.scss';
 
-class ProductAttributeControl extends Component {
-	constructor() {
-		super();
-
-		this.renderItem = this.renderItem.bind( this );
-		this.onExpandAttribute = this.onExpandAttribute.bind( this );
-	}
-
-	onExpandAttribute( item ) {
+const ProductAttributeControl = ( { attributes, error, expandedAttribute, onChange, onOperatorChange, isLoading, operator, selected, termsAreLoading, termsList } ) => {
+	const onExpandAttribute = ( item ) => {
```

Comment:
This should be declared outside of the component otherwise the `<SearchListItem />` will receive a new function on every re-render of this component.  One of the gotchas of switching to functional components from a class component.  It's okay when you're not passing the function through as a prop but it's being passed through as the value for the `onSelect` prop here.

Of course, this may be a moot point because there are other places where there will always be a new reference passed through as a prop (see [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/935/files/0f886d2cb432b5c3935a35725bdbf9fa05f1e016..107e163abc7314f26c79c5d8f87c3ba91d224863#diff-feb101de314b6e8ee6aa10b4a5e00111R86)) for example.  So if we're fine with `<SearchListItem />` _always_ re-rendering when `<ProductAttributeControl />` does, then we can just leave things as is I suppose.

Reply:
`onExpandAttribute` relies on the `onChange` prop, so we would need to re-create the function anyway on every re-render with the new `onChange`, no?

Reply:
Oh right! yup.